server/service/src/invoice/inbound_shipment/update/generate.rs

```diff
@@ -80,27 +84,17 @@ pub(crate) fn generate(
                 supplier_id: &update_invoice.name_link_id,
                 currency_id: update_invoice.currency_id.clone(),
                 currency_rate: &update_invoice.currency_rate,
-                default_donor_id: update_invoice.default_donor_id.clone(),
-                donor_update_method: patch.update_donor_method,
             },
         )?)
     } else {
         None
     };
 
     let location_movements = if let Some(batches) = &batches_to_update {
-        let generate_movement = batches
-            .iter()
-            .filter_map(|batch| {
-                batch
-                    .line
-                    .location_id
-                    .clone()
-                    .map(|_| generate_location_movements(store_id.to_owned(), batch))
-            })
-            .collect();
-
-        Some(generate_movement)
+        Some(generate_location_movements(
```

Comment:
Side quest... did some refactoring on the location movements when LineAndStockLine changed shape..